packages/adapter-drizzle/src/mysql/index.ts

```diff
@@ -0,0 +1,261 @@
+import { and, eq } from "drizzle-orm"
+import crypto from "node:crypto"
+import {
+  int,
+  timestamp,
+  mysqlTable,
+  primaryKey,
+  varchar,
+} from "drizzle-orm/mysql-core"
+import type { Adapter, AdapterAccount } from "@auth/core/adapters"
+import { MySql2Database } from "drizzle-orm/mysql2"
+
+/** @internal */
+export const users = mysqlTable("users", {
+  id: varchar("id", { length: 255 }).notNull().primaryKey(),
+  name: varchar("name", { length: 255 }),
+  email: varchar("email", { length: 255 }).notNull(),
+  emailVerified: timestamp("emailVerified", {
+    mode: "date",
+    fsp: 3,
+  }).defaultNow(),
+  image: varchar("image", { length: 255 }),
+})
+
+/** @internal */
+export const accounts = mysqlTable(
+  "accounts",
+  {
+    userId: varchar("userId", { length: 255 })
+      .notNull()
+      .references(() => users.id, { onDelete: "cascade" }),
+    type: varchar("type", { length: 255 })
+      .$type<AdapterAccount["type"]>()
+      .notNull(),
+    provider: varchar("provider", { length: 255 }).notNull(),
+    providerAccountId: varchar("providerAccountId", { length: 255 }).notNull(),
+    refresh_token: varchar("refresh_token", { length: 255 }),
+    access_token: varchar("access_token", { length: 255 }),
+    expires_at: int("expires_at"),
+    token_type: varchar("token_type", { length: 255 }),
+    scope: varchar("scope", { length: 255 }),
+    id_token: varchar("id_token", { length: 255 }),
+    session_state: varchar("session_state", { length: 255 }),
```

Comment:
I copied this code in my project because I wanted to start using Planetscale + Next-Auth + Drizzel. So thank you!. 

Is this mysql adapter the one we should use if we want to use Planetscale? If yes, in my case I had to use for `id_token` and `session_state` a `text` field not a `varchar` because I got an error when using Google Auth.

Reply:
Sorry, I didn't see your previous comment here https://github.com/nextauthjs/next-auth/pull/7165#discussion_r1263313311

Ignore this. I guess I'll use my custom adapter for planet scale for now. Thanks for the work!

Reply:
Would you mind sharing your adapter? I'm also using Planetscale so this bugs me as well ... 😕

Reply:
Hi @andresgutgon! It would be nice, maybe you have a public repo in which we can read your twist of the adapter?

Reply:
> I copied this code in my project because I wanted to start using Planetscale + Next-Auth + Drizzel. So thank you!. 
> 
> Is this mysql adapter the one we should use if we want to use Planetscale? If yes, in my case I had to use for `id_token` and `session_state` a `text` field not a `varchar` because I got an error when using Google Auth.
> 
> 
Sharing is caring

Reply:
Not much to it. I just copy & paste from this PR when there was the planetscale adapter and fixed the errors for Google Auth provider.
This is working for me: Google Auth + Planetscale
https://gist.github.com/andresgutgon/dcbf135f890450f0a03eeb9b7891b268

Reply:
I was just wondering since I was looking at the schema generated for Primsa, but shouldn't the `refresh_token` & `access_token` columns have a `text` type as well?

Reply:
In my case with Google Auth was not necessary. The problem with using text instead of varchar is that it takes more space in the db to store the column afaik. So i use text only when strictly necessary

Reply:
@andresgutgon  did you get prepared queries working?

Reply:
I don't even know what's that lol. So I guess no xD. What you see in the gist is what I did. Nothing else related with Drizzle + NextAuth + Planetscale

Reply:
@andresgutgon  prepared queries with your build are not erroring out locally. Pushing to Versel now

Reply:
Can you explain here quickly what's a prepared query and what's the benefit?

Reply:
A prepared statement is kinda what the name implies. Instead of generate each sql query on request which is the normal procedure usually when using ORMs, Drizzle can make prepared statements, where the SQL is made on build, and when requested by code it it can use it directly which improves performance a lot because it can reuse the binary compilation. It can also take in params if doing joins and what not, which still is much faster. See here for more https://orm.drizzle.team/docs/performance#

Reply:
so Prepared statement

> When you run query to the database there're several things that happens:
> 
> - all the configurations of the query builder got concatenated to the SQL string
>-  that string and params are sent to the database driver
>-  driver compiles SQL query to the binary SQL executable format and sends it to the database
>-  With prepared statements you do SQL concatenation once on the Drizzle ORM side and then database driver is able to reuse precompiled binary SQL instead of parsing query all the time. It has extreme performance benefits on large SQL queries.

https://orm.drizzle.team/docs/performance

can also watch the first half of this video
- [](https://www.youtube.com/watch?v=O9zxsSl9zgE&t=430s)